internal/resources/slo/resource_slo.go

```diff
@@ -57,17 +57,44 @@ Resource manages Grafana SLOs.
 						"freeform": &schema.Schema{
 							Type:     schema.TypeList,
 							MaxItems: 1,
-							Required: true,
+							Optional: true,
 							Elem: &schema.Resource{
 								Schema: map[string]*schema.Schema{
 									"query": &schema.Schema{
 										Type:        schema.TypeString,
-										Optional:    true,
+										Required:    true,
 										Description: "Freeform Query Field",
 									},
 								},
 							},
 						},
+						"ratio": &schema.Schema{
+							Type:     schema.TypeList,
```

Comment:
is this a list just because ratio contains more than one field? I'm a HCL noob, just curious. I almost thought it meant a list of ratio definitions.

Reply:
Yeah it's a TypeList because I wanted the nesting, but the list has a restriction of being only a length of one (hence the line `MaxItems: 1`. And then that singular item is a struct consisting of `success_metric`, `total_metric`, and `group_by_labels`